git_machete/cli.py

```diff
@@ -354,192 +505,149 @@ def check_required_param(in_args: List[str], allowed_values_string: str) -> str:
-        elif cmd in ("s", "status"):
-            expect_no_param(parse_options(args, "Ll",
-                                          ["color=", "list-commits-with-hashes", "list-commits",
-                                           "no-detect-squash-merges"]))
+            machete_client.squash(
+                current_branch,
+                cli_opts.opt_fork_point or machete_client.fork_point(
+                    current_branch, use_overrides=True))
+        elif cmd == "status":
             machete_client.read_definition_file()
             machete_client.expect_at_least_one_managed_branch()
             machete_client.status(warn_on_yellow_edges=True)
         elif cmd == "traverse":
-            traverse_long_opts = ["fetch", "list-commits", "merge",
-                                  "no-detect-squash-merges",
-                                  "no-edit-merge", "no-interactive-rebase",
-                                  "no-push", "no-push-untracked", "push", "push-untracked",
-                                  "return-to=", "start-from=", "whole", "yes"]
-            expect_no_param(parse_options(args, "FlMnWwy", traverse_long_opts))
-            if cli_opts.opt_start_from not in ("here", "root", "first-root"):
+            if cli_opts.opt_start_from not in {"here", "root", "first-root"}:
                 raise MacheteException(
-                    "Invalid argument for `--start-from`. Valid arguments: `here|root|first-root`.")
+                    "Invalid argument for `--start-from`. "
+                    "Valid arguments: `here|root|first-root`.")
             if cli_opts.opt_return_to not in ("here", "nearest-remaining", "stay"):
                 raise MacheteException(
-                    "Invalid argument for `--return-to`. Valid arguments: here|nearest-remaining|stay.")
+                    "Invalid argument for `--return-to`. "
+                    "Valid arguments: here|nearest-remaining|stay.")
             machete_client.read_definition_file()
             git.expect_no_operation_in_progress()
             machete_client.traverse()
         elif cmd == "update":
-            args1 = parse_options(args, "f:Mn", ["fork-point=", "merge", "no-edit-merge",
-                                                 "no-interactive-rebase"])
-            expect_no_param(args1, ". Use `-f` or `--fork-point` to specify the fork point commit")
             machete_client.read_definition_file()
             git.expect_no_operation_in_progress()
             machete_client.update()
         elif cmd == "version":
             version()
             sys.exit()
-        else:
-            short_usage()
-            raise MacheteException(
-                f"\nUnknown command: `{cmd}`. Use `git machete help` to list possible commands")
 
-    except getopt.GetoptError as e:
-        short_usage()
+    except (argparse.ArgumentError, argparse.ArgumentTypeError) as e:
+        print(short_usage())
         sys.stderr.write(f"\n{e}\n")
```

Comment:
While you're here... pls remove the newline here, it usually look weird:
```suggestion
        sys.stderr.write(f"{e}\n")
```